### ISSUE TITLE
Conv2D not working in MPC

### ISSUE THREAD

**Issue by avitalsh** [OpenMined/PySyft#2117]
Hi,
I tried running the code in [tutorial 10](https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/Part%2010%20-%20Secure%20Deep%20Learning%20Classification.ipynb) with only changing the model to be a conv2D layer, followed by a fc layer:

```
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv = nn.Conv2d(1, 3, (3,3), bias=False)
        self.fc = nn.Linear(3*26*26, 10, bias=False)

    def forward(self, x):
        x = self.conv(x)
        x = x.view(-1, 3*26*26)
        x = self.fc(x)
        return x

```
I got the following error - 
RuntimeError: _thnn_conv2d_forward is not implemented for type torch.LongTensor

I assume it has to do with the fact that conv2D supports float tesnsors, where SMPC works on integers (so we use .fix_prec()).

Is there a way to solve this issue and to perform secure inference with convolutional neural networks? 

Thanks

**Comment by mari-linhares** [comment]
Hey @avitalsh,

> I assume it has to do with the fact that conv2D supports float tensors, where SMPC works on integers (so we use .fix_prec()).

I've just tried and I think this is the case as well..

> Is there a way to solve this issue and to perform secure inference with convolutional neural networks?

Yes, feel free to create a PR!

**Comment by Jasopaum** [comment]
I am dealing with this issue

**Comment by mari-linhares** [comment]
Fixed by #2165
